src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacFieldAndReferencedFieldMismatch/DacFieldAndReferencedFieldMismatchAnalyzer.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Acuminator.Analyzers.StaticAnalysis.Dac;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.DiagnosticSuppression;
+using Acuminator.Utilities.Roslyn.PXFieldAttributes;
+using Acuminator.Utilities.Roslyn.PXFieldAttributes.Enum;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Acuminator.Utilities.Roslyn.Semantic.Attribute;
+using Acuminator.Utilities.Roslyn.Semantic.Dac;
+using Acuminator.Utilities.Roslyn.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Acuminator.Analyzers.StaticAnalysis.DacFieldAndReferencedFieldMismatch
+{
+	public class DacFieldAndReferencedFieldMismatchAnalyzer : DacAggregatedAnalyzerBase
+	{
+		private readonly Dictionary<INamedTypeSymbol, DacSemanticModel?> _modelsCache = new(SymbolEqualityComparer.Default);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(
+				Descriptors.PX1078_TypesOfDacFieldAndReferencedFieldMismatch,
+				Descriptors.PX1078_TypesOfDacFieldAndReferencedFieldHaveDifferentSize);
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, DacSemanticModel dac)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var propertiesToCheck = dac.DeclaredDacFieldPropertiesWithAcumaticaAttributes
+				.Select(p => (p, TryGetForeignReferenceAttribute(pxContext, p)))
+				.Where(p => p.Item2 is not null);
+
+			var fieldTypeAttributesMetadataProvider = new FieldTypeAttributesMetadataProvider(pxContext);
+
+			foreach (var (property, foreignReferenceAttribute) in propertiesToCheck)
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				CheckProperty(context, property, foreignReferenceAttribute!, dac, pxContext, fieldTypeAttributesMetadataProvider);
+			}
+		}
+
+		private static DacFieldAttributeInfo? TryGetForeignReferenceAttribute(PXContext pxContext, DacPropertyInfo property) =>
+			property.Attributes.FirstOrDefault(x => ContainsAttributeMatching(x, pxContext.AttributeTypes.PXSelectorAttribute.Type));
+
+		private void CheckProperty(SymbolAnalysisContext context, DacPropertyInfo property, DacFieldAttributeInfo foreignReferenceAttribute,
+			DacSemanticModel dac, PXContext pxContext, FieldTypeAttributesMetadataProvider fieldTypeAttributesMetadataProvider)
+		{
+			var foreignField = ExtractForeignFieldSymbol(foreignReferenceAttribute, pxContext);
+
+			if (foreignField is null)
+				return;
+
+			var parentType = (foreignField.ContainingSymbol as INamedTypeSymbol)!;
```

Comment:
Not insisting, but containing type should be generally used to get the declaring type.
```suggestion
			var parentType = foreignField.ContainingType!;
```